File: go/models/escalations.go
package models

import "time"

type Escalation struct {
	ID          string    `json:"id"`
	JudgmentID  string    `json:"judgment_id"`
	Reason      string    `json:"reason"`
	Severity    string    `json:"severity"`
	Status      string    `json:"status"`
	AssignedTo  string    `json:"assigned_to,omitempty"`
	Resolution  string    `json:"resolution,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	ResolvedAt  *time.Time `json:"resolved_at,omitempty"`
}

type EscalationCreateRequest struct {
	JudgmentID string `json:"judgment_id"`
	Reason     string `json:"reason"`
	Severity   string `json:"severity,omitempty"`
}

type EscalationResolveRequest struct {
	Resolution string `json:"resolution"`
	Action     string `json:"action,omitempty"`
}

type EscalationAssignRequest struct {
	AssignedTo string `json:"assigned_to"`
}

type EscalationStats struct {
	Total      int            `json:"total"`
	Open       int            `json:"open"`
	Resolved   int            `json:"resolved"`
	AvgResolve string         `json:"avg_resolve_time"`
	BySeverity map[string]int `json:"by_severity"`
}

File: rust/src/models/escalations.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct Escalation {
    pub id: String,
    pub judgment_id: String,
    #[serde(default)]
    pub status: String,
    #[serde(default)]
    pub reason: String,
    #[serde(default)]
    pub priority: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub assigned_to: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub resolution: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub resolved_at: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct EscalationCreateRequest {
    pub judgment_id: String,
    pub reason: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub priority: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct EscalationResolveRequest {
    pub resolution: String,
    pub final_decision: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct EscalationAssignRequest {
    pub assigned_to: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct EscalationStats {
    #[serde(default)]
    pub total: i64,
    #[serde(default)]
    pub pending: i64,
    #[serde(default)]
    pub in_progress: i64,
    #[serde(default)]
    pub resolved: i64,
    #[serde(default)]
    pub avg_resolution_time_ms: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct EscalationListResponse {
    pub escalations: Vec<Escalation>,
    #[serde(default)]
    pub total: i64,
    #[serde(default)]
    pub page: i32,
    #[serde(default)]
    pub limit: i32,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct EscalationListParams {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub status: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub page: Option<i32>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub limit: Option<i32>,
}
